### PR TITLE
feat: add auth_token in tableau config

### LINKED ISSUE #358: feat: add `auth_token` to tableau extractor
**Is your feature request related to a problem? Please describe.**
We use `sessionID` for setting up client in metabase, so similary we can use `auth token` for the purpose on the tableau.

**Describe the solution you'd like**
Add `auth token` to the config for tableau, and make username and password optional. Similar to #351

### PR DESCRIPTION
**Pull request by GrayFlash**
closes #358

### CODE REVIEW

File: plugins/extractors/tableau/tableau_test.go [hunk 1]
```diff
@@ -36,6 +36,17 @@ func TestInit(t *testing.T) {
 
 		assert.Equal(t, plugins.InvalidConfigError{}, err)
 	})
+	t.Run("should return no error for valid config", func(t *testing.T) {
```
Comment (StewartJingga) [inline plugins/extractors/tableau/tableau_test.go:39]: can you add an extra test similar to this 🙏 
https://github.com/odpf/meteor/blob/main/plugins/extractors/metabase/metabase_test.go#L38-L64

**Review by StewartJingga — CHANGES_REQUESTED** [review]

File: plugins/extractors/tableau/tableau_test.go (hunk 1 above)
Comment (GrayFlash) [reply plugins/extractors/tableau/tableau_test.go:39]: hey, I already added a test for the validation right, and in client I have made changes to logic where if auth_token is present, then it skips generating one. Can you let me know what scope should I test it for??

Comment (GrayFlash) [reply plugins/extractors/tableau/tableau_test.go]: @StewartJingga added few more tests with various combinations.

File: plugins/extractors/tableau/tableau.go [hunk 2]
```diff
@@ -35,8 +35,10 @@ type Config struct {
 	Host       string `mapstructure:"host" validate:"required"`
 	Version    string `mapstructure:"version" validate:"required"` // float as string
 	Identifier string `mapstructure:"identifier" validate:"required"`
-	Username   string `mapstructure:"username" validate:"required"`
-	Password   string `mapstructure:"password" validate:"required"`
+	Username   string `mapstructure:"username"`
+	Password   string `mapstructure:"password" validate:"required_with=Username"`
+	AuthToken  string `mapstructure:"auth_token" validate:"required_without=Username"`
+	SiteID     string `mapstructure:"site_id" validate:"required_without=Username"`
```
Comment (GrayFlash) [inline plugins/extractors/tableau/tableau.go:41]: @StewartJingga I have also updated the config struct here with the required conditions.

`username` is optional, and whenever present now requires a `password`. and whenever absent requires `auth_token` and `site_id`.

**Review by StewartJingga — APPROVED** [review]

**Pull request merged**